src/box/vy_mem.c

```diff
@@ -114,9 +115,12 @@ vy_mem_new(struct vy_mem_env *env, struct key_def *cmp_def,
 	index->space_cache_version = space_cache_version;
 	index->format = format;
 	tuple_format_ref(format);
+	matras_allocator_create(&index->matras_allocator,
+				VY_MEM_TREE_EXTENT_SIZE,
+				vy_mem_tree_extent_alloc,
+				vy_mem_tree_extent_free);
 	vy_mem_tree_create(&index->tree, cmp_def,
-			   vy_mem_tree_extent_alloc,
-			   vy_mem_tree_extent_free, index, NULL);
+			   &index->matras_allocator, NULL);
```

Comment:
Technically, I think we could use the global generation counter for allocations from a `vy_mem` because only the newest `vy_mem` is active (the ones that use older generations are sealed for modifications). Moreover, allocating a block with an older generation counter would trigger an assertion in `lsregion_alloc`. I'll see what we can do about it. For now, I guess we can leave it as is.

BTW we must never reserve blocks with this allocator because it's garbage-collected externally. Seems fine because we don't use matras read views in Vinyl.